### PR TITLE
Block Supports: Add panel specific className

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Related:
- https://github.com/WordPress/gutenberg/pull/33744
- https://github.com/WordPress/gutenberg/pull/34027
- https://github.com/WordPress/gutenberg/pull/33743

## Description

This adds a `className` specific to each grouped inspector controls slot. It allows different block supports to override some component styles within the `ToolsPanel` until those components are migrated to the new context system.

## How has this been tested?

Manually.

1. Loaded a post in the block editor that has a cover block and selected that block.
2. Ensured that the dimensions panel had the `dimensions-block-support-panel` class.
3. Tested this with rebased versions of the related PRs linked above.

## Screenshots 

Screen Shot 2021-10-20 at 5 47 30 pm

## Types of changes
Enhancement.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by ramonjd** [comment]
Tests well for dimension and border. I couldn't cleanly rebase or merge this PR on top of Color Block Support: Switch to ToolsPanel for displaying UI #34027, or the other way around unfortunately. It looks like that PR needs a rebase itself.

So wasn't able to reliably test the color panel. Seeing as it's not yet merged I guess it's okay?

Screen Shot 2021-10-20 at 7 38 53 pm

Conceptually, I think adding specific classnames for grouped ToolsPanels is okay, seeing as there's a [precedent](https://github.com/WordPress/gutenberg/blob/v11.7.0/packages/block-editor/src/hooks/border.js#L48) for passing custom `className`s to `<PanelBody />` for example.

Is it something that we'd continue to want to do in the future? I can't tell. I don't think the desire to customize the internals of individual panels is going to go away though.

**Review by ramonjd — APPROVED** [review]
Tests well on for Dimensions and Border.  

I can see the className on the containing div.

I couldn't test Color, but seeing as it hasn't yet been migrated to the ToolsPanel, I'm assuming it should be fine (?)

File: packages/block-editor/src/components/inspector-controls/block-support-tools-panel.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ export default function BlockSupportToolsPanel( { children, label } ) {
 
 	return (
 		<ToolsPanel
+			className={ `${ group }-block-support-panel` }
```
Comment (ramonjd) [inline packages/block-editor/src/components/inspector-controls/block-support-tools-panel.js:49]: Is there any reason why `'group'` wouldn't be a string or empty?

Just wondering if you need a default or a conditional application of the className here.

Comment (aaronrobertshaw) [reply packages/block-editor/src/components/inspector-controls/block-support-tools-panel.js:49]: > Is there any reason why 'group' wouldn't be a string or empty?

There shouldn't be. It's the key used to [identify the inspector controls slot](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inspector-controls/slot.js#L20-L34). If no slot matches it never makes it to creating the block support tools panel. So `group` here should always be a known quantity from the [defined groups](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inspector-controls/groups.js).

**Comment by aaronrobertshaw** [comment]
> I couldn't cleanly rebase or merge this PR on top of Color Block Support: Switch to ToolsPanel for displaying UI #34027, or the other way around unfortunately. It looks like that PR needs a rebase itself.

Between when #34027 was created and now a lot has changed. I have rebased it locally and am working on resolving conflicts. In particular, the Global Styles sidebar has changed plenty. It now uses an updated navigation component as well as a new approach to getting and setting global styles. 

The same goes for the other block support panel PRs switching to the `ToolsPanel`.

I'm not sure that these panels in the global styles need to be updated now but I'm guessing consistency between the global styles and block editor inspector controls is still desirable. 

Either way, given we shouldn't target hardcoded class names directly within the `ToolsPanel` component. We need a means of styling components that aren't yet hooked up to the context system. Tweaking those components directly isn't an option given they can also be being used as ad hoc controls in other panels. 

> Is it something that we'd continue to want to do in the future? I can't tell. I don't think the desire to customize the internals of individual panels is going to go away though.

Even when our components are updated to leverage the context system and Emotion for styling, I believe there'll be value in being able to target these panels via CSS. While still experimental, soon enough, we'll be allowing controls to be injected into these block support panels via SlotFill. I can definitely see those ad hoc controls benefiting from the new class names.

**Comment by aaronrobertshaw** [comment]
To help progress the other PRs switching the block supports to the `ToolsPanel`, I'll merge this one.

**Pull request merged**